src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -274,32 +327,34 @@ public static void AssertValidVariableValue(ISchema schema, IGraphType type, obj
         }
 
         /// <summary>
-        /// Coerces a variable value to a compatible .NET type for the variable's graph type.
+        /// Coerces a literal value to a compatible .NET type for the variable's graph type.
+        /// Typically this is a value for a field argument or default value for a variable.
         /// </summary>
-        public static object CoerceValue(ISchema schema, IGraphType type, IValue input, Variables variables = null, object fieldDefault = null)
+        public static ArgumentValue CoerceValue(ISchema schema, IGraphType type, IValue input, Variables variables = null, object fieldDefault = null)
         {
             if (type is NonNullGraphType nonNull)
             {
-                // validation rules and/or AssertValidVariableValue have verified that this is not null
+                // validation rules have verified that this is not null
```

Comment:
But they still can be disabled.

Reply:
Yeah I noticed how this worked. I figured that if you disabled the rule then presumably you didn’t want the check.

Reply:
This consideration in the same form should be written in the comments so as not to mislead the reader.

Reply:
```suggestion
                // validation rules have verified that this is not null; if the validation rule was not executed, it
                // is assumed that the caller does not wish this check to be executed
```

Reply:
Commit if this looks okay to you.